hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -404,6 +404,11 @@ public static InetSocketAddress createSocketAddr(String target) {
   private final StorageLocationChecker storageLocationChecker;
 
   private final DatasetVolumeChecker volumeChecker;
+  volatile FsDatasetSpi<? extends FsVolumeSpi> allData = null;
+  private Thread checkDiskThread = null;
+  private final int checkDiskInterval = 5*1000;
+  private Object checkDiskMutex = new Object();
+  private List<StorageLocation> errorDisk = null;
```

Comment:
`allData = data + errorDisk` right? It seems that `allData` is redundant variable.

Reply:
`allData = data + errorDisk` is right. `allData` will use in `volumeChecker.checkAllVolumes(allData)`.
If no `allData`, we must make a variable before `volumeChecker.checkAllVolumes`, the variable is `data + errorDisk`. So I think `allData` is ok .